### DISCUSSION TITLE
how to periodically update segment for K8s info output ?

### DISCUSSION THREAD

**Discussion by gkoenig** [erikw/tmux-powerline#282] [Q&A] [answered]
Hi,
I've written a small, simple script which outputs the _k8s namespace_ and _k8s server address_ to a segment, based on the output of command `kubectl config view`
It works great and I can see the desired output in the segment of the status bar.
The problem is, as soon as I change my $KUBECONFIG variable in the shell, the output in the segment shall update as well.....because changing this env variable also changes the output of the `kubectl config view` command.

How can I trigger the segment update periodically, so that it always reflects the current output of the kubectl command ? ....at the moment it doesn't update, it always shows the info of the time tmux was started.
In my .tmux.conf there is the _status-interval_ option set =>
`set-option -g status-interval 2`

Thanks !

**Comment by erikw** [comment, marked as answer]
The way environmental variables and shells works is that a change of an envvar is local to the current shell. tmux-powerline works in a different shell than the one you changed `$KUBECONFIG` in. Thus, there must be some way to inform tmux-powerline on which value of `$KUBECONFIG` should be used.

Two ideas to solve this would be:
1. Let your segment read the value from a state file like `~/.kubeconfig`. Instead of changing the envvar in your shell, you write the current value to a file. To make this easy, you can make a shell function that you source or include in your shell startup file like
   *
   ```bash
   kubeconfig() {
    local value="$1"
    export KUBECONFIG="$value"
    echo "$value" > $HOME/.kubeconfig
   }
   ```
   * Then in your shell, change the variable with `$ kubeconfig new-value-of-envvar`
   * Then in your segment just read this value 
   ```bash
   export KUBECONFIG=$(cat ~/.kubeconfig)
   ```
1. Each time the prompt is rendered, copy the local `$KUBECONFIG` and send it to the tmux envvar with `tmux setenv`. This is documented at https://github.com/erikw/tmux-powerline#for-tmux-versions--21

**Reply by gkoenig** [reply]
Many thanks @erikw for those hints.
I went with option 1. => works great :D

**Reply by erikw** [reply]
Great, I'm glad I could help!
